src/operations/SymbolicOperation.cpp

```diff
@@ -63,43 +72,48 @@ namespace qc {
 
         return U3;
     }
-    OpType SymbolicOperation::parseU3(fp& lambda, fp& phi,
-                                      [[maybe_unused]] const Symbolic& theta) {
-        if (std::abs(lambda) < PARAMETER_TOLERANCE) {
-            lambda = 0.L;
-            if (std::abs(phi) < PARAMETER_TOLERANCE) {
-                phi = 0.L;
-            }
-        }
-
-        if (std::abs(lambda - PI_2) < PARAMETER_TOLERANCE) {
-            lambda = PI_2;
-
-            if (std::abs(phi - PI_2) < PARAMETER_TOLERANCE) {
-                phi = PI_2;
-            }
+    OpType SymbolicOperation::parseU3(fp& theta, fp& phi,
+                                      [[maybe_unused]] const Symbolic& lambda) {
+        if (std::abs(theta) < PARAMETER_TOLERANCE &&
+            std::abs(phi) < PARAMETER_TOLERANCE) {
+            phi   = 0.L;
+            theta = 0.L;
+            return SymbolicOperation::parseU1(lambda);
         }
 
-        if (std::abs(lambda - PI) < PARAMETER_TOLERANCE) {
-            lambda = PI;
-            if (std::abs(phi) < PARAMETER_TOLERANCE) {
-                phi = 0.L;
-            }
+        if (std::abs(theta - PI_2) < PARAMETER_TOLERANCE) {
+            theta = PI_2;
+            return parseU2(phi, lambda);
         }
+        // parse a real u3 gate
+        checkInteger(phi);
+        checkFractionPi(phi);
+        checkInteger(theta);
+        checkFractionPi(theta);
 
+        return U3;
+    }
+    OpType SymbolicOperation::parseU3([[maybe_unused]] const Symbolic& theta,
+                                      [[maybe_unused]] const Symbolic& phi, fp& lambda) {
         // parse a real u3 gate
         checkInteger(lambda);
         checkFractionPi(lambda);
+
+        return U3;
+    }
+    OpType SymbolicOperation::parseU3([[maybe_unused]] const Symbolic& theta, fp& phi,
+                                      [[maybe_unused]] const Symbolic& lambda) {
+        // parse a real u3 gate
         checkInteger(phi);
         checkFractionPi(phi);
 
         return U3;
     }
-    OpType SymbolicOperation::parseU3(const Symbolic& lambda,
-                                      const Symbolic& phi, fp& theta) {
+    OpType SymbolicOperation::parseU3(fp& theta, [[maybe_unused]] const Symbolic& phi,
+                                      [[maybe_unused]] const Symbolic& lambda) {
         if (std::abs(theta - PI_2) < PARAMETER_TOLERANCE) {
             theta = PI_2;
-            return parseU2(lambda, phi);
+            return parseU2(phi, lambda);
         }
```

Comment:
Do `phi` and `lambda` really need `[[maybe_unused]]`?

Reply:
You are right. Removed in https://github.com/cda-tum/qfr/pull/280/commits/c2077d9819a283d7376abf10a6511db3c0886ab6.